AdyenCardScanner/Sources/CardImageParser.swift

```diff
@@ -0,0 +1,185 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import CoreImage.CIFilterBuiltins
+import Foundation
+import Vision
+
+protocol CardImageParsing {
+    func parse(
+        image: CIImage,
+        completion: @escaping (CreditCard) -> Void
+    )
+}
+
+class CardImageParser: CardImageParsing {
+
+    private enum Constants {
+        static let expirationDateRegex = "\\d{2}\\/\\d{2,4}"
+        static let topCandidates = 10
+
+        static let cardNumberConfidence: Float = 0.4
+        static let expirationDateConfidence: Float = 0.4
+    }
+
+    // MARK: - Properties
+
+    private let expirationDateFormatter: ExpirationDateFormatting
+    private var cachedCardNumber: String?
+    private var cachedExpirationDate: Date?
+
+    // MARK: - Initializers
+
+    init(expirationDateFormatter: ExpirationDateFormatting) {
+        self.expirationDateFormatter = expirationDateFormatter
+    }
+
+    // MARK: - CardImageParsing
+
+    func parse(
+        image: CIImage,
+        completion: @escaping (CreditCard) -> Void
+    ) {
+        guard let transformedImage = transform(image: image) else { return }
+
+        let recognizeTextRequest = VNRecognizeTextRequest()
+        recognizeTextRequest.recognitionLevel = .accurate
+        recognizeTextRequest.usesLanguageCorrection = false
+
+        let imageRequestHandler = VNImageRequestHandler(ciImage: transformedImage, options: [:])
+        try? imageRequestHandler.perform([recognizeTextRequest])
```

Comment:
Maybe better put it in try/catch instead or make `parse(image: completion)` throws?